src/test/java/RebindTest.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    @Test
+    public void testRebindMultipleConnections() {
+        RelpConfig relpConfig = new RelpConfig();
+
+        // Multiple connections together using the RelpConnectionFactory and the Pool
+        RelpConnectionFactory relpConnectionFactory = new RelpConnectionFactory(relpConfig);
+        Pool<IManagedRelpConnection> pool = new Pool<>(relpConnectionFactory, new ManagedRelpConnectionStub());
+
+        IManagedRelpConnection firstConnection = pool.get();
+        IManagedRelpConnection secondConnection = pool.get();
+        IManagedRelpConnection thirdConnection = pool.get();
+
+        SyslogMessage syslogMessage = new SyslogMessage()
+                .withFacility(Facility.USER)
+                .withSeverity(Severity.INFORMATIONAL)
+                .withMsg("foobar");
+
+        byte[] bytes = syslogMessage.toRfc5424SyslogMessage().getBytes(StandardCharsets.UTF_8);
+
+        for (int i = 0; i <= relpConfig.rebindRequestAmount * 3; i++) { // 3 rebinds
+            firstConnection.ensureSent(bytes);
+            pool.offer(firstConnection);
+            secondConnection.ensureSent(bytes);
+            pool.offer(secondConnection);
+            thirdConnection.ensureSent(bytes);
+            pool.offer(thirdConnection);
+        }
+
+        List<CountingFrameDelegate> frameDelegates = relpServer.frameDelegates();
+        Assertions.assertEquals(12, frameDelegates.size());
+
+        for (int i = 0; i < 9; i++) {
+            Assertions.assertEquals(relpConfig.rebindRequestAmount, frameDelegates.get(i).recordsReceived());
+        }
+
+        for (int i = 9; i < 12; i++) {
+            Assertions.assertEquals(1, frameDelegates.get(i).recordsReceived()); // last one receives only 1 message
+        }
+    }
+
+    @Test
+    public void testRebindDisabled() {
+        System.setProperty("relp.rebindEnabled", "false");
+
+        RelpConfig relpConfig = new RelpConfig();
+
+        RelpConnectionFactory relpConnectionFactory = new RelpConnectionFactory(relpConfig);
+        Pool<IManagedRelpConnection> pool = new Pool<>(relpConnectionFactory, new ManagedRelpConnectionStub());
+
+        IManagedRelpConnection connection = pool.get();
+
+        SyslogMessage syslogMessage = new SyslogMessage()
+                .withFacility(Facility.USER)
+                .withSeverity(Severity.INFORMATIONAL)
+                .withMsg("foobar");
+
+        for (int i = 0; i <= relpConfig.rebindRequestAmount; i++) { // rebind + 1
+            connection.ensureSent(syslogMessage.toRfc5424SyslogMessage().getBytes(StandardCharsets.UTF_8));
+        }
+
+        // Only one connection should have been used with rebindEnabled = false
+        Assertions.assertEquals(1, relpServer.frameDelegates().size());
+        Assertions
+                .assertEquals(relpConfig.rebindRequestAmount + 1, relpServer.frameDelegates().get(0).recordsReceived());
+    }
+
```

Comment:
Maybe add a test to check that rebind throws when you give < 1 value

Reply:
Added tests for -1 and 0